src/config.ts

```diff
@@ -23,15 +23,59 @@ export interface Config {
   fetch?: (input: string, init?: Req) => Promise<Res>
   arrayMode?: boolean
   fullResult?: boolean
+  decode?: DecodeConfig
 }
 
 export interface ExecuteOptions {
   arrayMode?: boolean
   fullResult?: boolean
+  decode?: DecodeConfig
 }
 
 export interface TxOptions {
   isolation?: 'READ COMMITTED' | 'REPEATABLE READ'
 }
 
 export type ExecuteArgs = object | any[] | null
+
+export const enum ColumnType {
+  TINYINT = 'TINYINT',
+  UNSIGNED_TINYINT = 'UNSIGNED TINYINT',
+  SMALLINT = 'SMALLINT',
+  UNSIGNED_SMALLINT = 'UNSIGNED SMALLINT',
+  MEDIUMINT = 'MEDIUMINT',
+  INT = 'INT',
+  UNSIGNED_INT = 'UNSIGNED INT',
+  YEAR = 'YEAR',
+  FLOAT = 'FLOAT',
+  DOUBLE = 'DOUBLE',
+  BIGINT = 'BIGINT',
+  UNSIGNED_BIGINT = 'UNSIGNED BIGINT',
+  DECIMAL = 'DECIMAL',
+  CHAR = 'CHAR',
+  VARCHAR = 'VARCHAR',
+  BINARY = 'BINARY',
+  VARBINARY = 'VARBINARY',
+  TINYTEXT = 'TINYTEXT',
+  TEXT = 'TEXT',
+  MEDIUMTEXT = 'MEDIUMTEXT',
+  LONGTEXT = 'LONGTEXT',
+  TINYBLOB = 'TINYBLOB',
+  BLOB = 'BLOB',
+  MEDIUMBLOB = 'MEDIUMBLOB',
+  LONGBLOB = 'LONGBLOB',
+  DATE = 'DATE',
+  TIME = 'TIME',
+  DATETIME = 'DATETIME',
+  TIMESTAMP = 'TIMESTAMP',
+  BIT = 'BIT',
+  JSON = 'JSON'
+}
+
+export type DecodeConfig = Partial<
+  {
+    [P in ColumnType]: (rawValue: string) => any
+  } & {
+    [key: string]: (rawValue: string) => any
```

Comment:
why we need `& {[key: string]: (rawValue: string) => any}`

Reply:
for future unlisted field types (like `VECTOR`)

Reply:
I think 
```
{
  [P in ColumnType]?: (rawValue: string) => any
} & {
  [key: string]: (rawValue: string) => any
}
```
is equal to 

```
  key: string]: (rawValue: string) => any
```

Both of them can't limit the type of key in typescript. I prefer to use one of `[P in ColumnType]?: (rawValue: string) => any` and `[key: string]: (rawValue: string) => any`

Reply:
The type system in typescript aims to check code in compile time.
With `[P in ColumnType]`, we will meet the compile error with the key not included in ColumnType. However `[key: string]` breaks this. This loses the meaning of defining ColumnType.

If we want to support other types. Just add them in ColumnType. (By the way, I think you can also pass any other value in js)

Reply:
```ts
{
  [P in ColumnType]?: (rawValue: string) => any
} & {
  [key: string]: (rawValue: string) => any
}
```

This one supports both IDE auto completion (DX) and compiler check for other types, but it's OK to not support custom column types (maybe never needed).

I'll keep just the first half.